CI builds fail in forked repos

**Describe the bug**

In forked repos without the RELEASE_REPO_TOKEN secret defined, CI builds fail attempting to upload release artifacts.

**To Reproduce**
You can see a failed build here: https://github.com/tfmorris/OpenRefine/runs/579874494?check_suite_focus=true

**Current Results**
Fails with the error `##[error]request to https://api.github.comnull/ failed, reason: getaddrinfo ENOTFOUND api.github.comnull` during the Upload Release Asset Windows setp.

**Expected behavior**
It should gracefully skip the upload step and not fail the build.

p.s. As an aside, it appears that the "nightly" build is actually built and uploaded for every commit, which seems wasteful.

--- a/.github/workflows/maven.yml
+++ b/.github/workflows/maven.yml
@@ -39,6 +39,7 @@ jobs:
       run: ./refine dist ${{ env.OR_VERSION }}
 
     - name: Create Release
+      if: github.repository == 'OpenRefine/OpenRefine'
       id: create_release
       run: echo ::set-env name=API_RELEASE::$(./.github/workflows/release_manager.sh)
       env:
@@ -48,6 +49,7 @@ jobs:
 
     - name: Upload Release Asset Windows
       id: upload-release-asset-win
+      if: github.repository == 'OpenRefine/OpenRefine'
       uses: actions/upload-release-asset@v1.0.1
       env:
         GITHUB_TOKEN: ${{ secrets.RELEASE_REPO_TOKEN }}
@@ -59,6 +61,7 @@ jobs:
 
     - name: Upload Release Asset Linux
       id: upload-release-asset-linux
+      if: github.repository == 'OpenRefine/OpenRefine'
       uses: actions/upload-release-asset@v1.0.1
       env:
         GITHUB_TOKEN: ${{ secrets.RELEASE_REPO_TOKEN }}
@@ -70,6 +73,7 @@ jobs:
 
     - name: Upload Release Asset Mac
       id: upload-release-asset-mac
+      if: github.repository == 'OpenRefine/OpenRefine'
       uses: actions/upload-release-asset@v1.0.1
       env:
         GITHUB_TOKEN: ${{ secrets.RELEASE_REPO_TOKEN }}

